lib/storage/azure.js

```diff
@@ -183,54 +183,53 @@ module.exports = function() {
     },
 
     copyIn: function(localPath, _path, options, callback) {
+      const randStr = () => Math.random().toString(36).substring(7);
       const path = _path[0] === '/' ? _path.slice(1) : _path;
-      const tmpFileName = Math.random().toString(36).substring(7);
-
-      // store gzipped file in temp dir
-      fs.mkdtemp(this.options.tempPath, (e, dir) => {
-        if (e) {
-          console.log(e);
-          return callback(e);
-        } else {
-          const tempPath = dir + '/' + tmpFileName;
-          let inp = fs.createReadStream(localPath);
-          let out = fs.createWriteStream(tempPath);
-
-          inp.on('error', e => {
-            console.log("Error in write stream", e);
-            callback(e);
-          });
+      const tmpFileName = randStr();
+      const tempPath = this.options.tempPath + '/' + tmpFileName;
+
+      let inp = fs.createReadStream(localPath);
+      let out = fs.createWriteStream(tempPath);
+      let hasError = false;
 
-          out.on('error', e => {
-            console.log("Error in write stream", e);
-            callback(e);
+      inp.on('error', inpErr => {
+        console.log("Error in read stream", inpErr);
```

Comment:
You mentioned "checkpoint", I guess you're not quite ready for review on this? I reckon that must be the case since you've only set hasError in one of the two cases and are not checking it yet, etc. I'll stop here.